bdb/tran.c

```diff
@@ -1430,7 +1430,8 @@ static int update_logical_redo_lsn(void *obj, void *arg)
 
     last = LISTC_BOT(&bdb_state->sc_redo_list);
     /* Add in order */
-    if (!last || log_compare(&last->lsn, &redo->lsn) <= 0)
+    /* 'Logical commits' can be out-of-order for prepared txns */
+    if (!last || (log_compare(&last->lsn, &redo->lsn) <= 0) || tran->is_prepared)
```

Comment:
Why does this work .. ?  Is it because the trn_repo_mutex guarantees that it will still be in physical commit order?  Give Rivers answer + bit more context ..